### PR TITLE
bg_iknin_susceil OK

### PR DESCRIPTION
**Pull request by kyleburnette**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

This was quite an actor to decomp. So many little things went wrong during this one.

D_80C0B0E4 and D_80C0B0E8 do NOT appear to be a Vec3f. If I make them a Vec3f, the function that uses them gets significantly dumber to match.

### CODE REVIEW

File: src/overlays/actors/ovl_Bg_Iknin_Susceil/z_bg_iknin_susceil.h [hunk 1]
```diff
@@ -3,11 +3,20 @@
...
 typedef struct BgIkninSusceil {
-    /* 0x000 */ Actor actor;
-    /* 0x144 */ char unk_144[0x28];
+    /* 0x000 */ DynaPolyActor dyna;
+    /* 0x15C */ AnimatedMaterial* animatedTexture;
+    /* 0x160 */ BgIkninSusceilActionFunc actionFunc;
+    /* 0x164 */ s16 timer;
+    /* 0x166 */ s16 unk166;
+    /* 0x168 */ s16 unk168;
+    /* 0x16A */ UNK_TYPE2 unk16A;
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Bg_Iknin_Susceil/z_bg_iknin_susceil.h:19]: This seems to be padding. You should be able to remove it and still match.

File: src/overlays/actors/ovl_Bg_Iknin_Susceil/z_bg_iknin_susceil.c [hunk 2]
```diff
@@ -21,46 +35,228 @@ const ActorInit Bg_Iknin_Susceil_InitVars = {
...
+};
+
+s32 func_80C0A740(BgIkninSusceil* this, GlobalContext* globalCtx) {
+    s32 pad2[2];
+    Vec3f offset;
+    ActorPlayer* player = PLAYER;
+
+    Actor_CalcOffsetOrientedToDrawRotation(&this->dyna.actor, &offset, &player->base.world.pos);
+
+    return ((D_80C0B0E8.x < offset.z) && (offset.z < D_80C0B0E8.y) && (offset.x > -240.0f) && (offset.x < D_80C0B0E4));
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Bg_Iknin_Susceil/z_bg_iknin_susceil.c:56]:
```suggestion
    return (D_80C0B0E8.x < offset.z) && (offset.z < D_80C0B0E8.y) && (offset.x > -240.0f) && (offset.x < D_80C0B0E4);
```

File: src/overlays/actors/ovl_Bg_Iknin_Susceil/z_bg_iknin_susceil.c [hunk 3]
```diff
@@ -21,46 +35,228 @@ const ActorInit Bg_Iknin_Susceil_InitVars = {
...
+    ActorPlayer* player = PLAYER;
+    Vec3f offset;
+    f32 temp1, temp2, temp3, temp4;
+
+    Actor_CalcOffsetOrientedToDrawRotation(&this->dyna.actor, &offset, &player->base.world.pos);
+    for (i = 0; i < 7; i++) {
+        temp3 = (D_80C0B0F0[i] * 80.0f) + 0.5f;
+        temp4 = (D_80C0B0F0[i] * 80.0f) + 79.5f;
+        if (1) {}
+        new_var = D_80C0B0F8[i] * (-80.0f); // temp seems to actually be needed for matching
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Bg_Iknin_Susceil/z_bg_iknin_susceil.c:100]: Are the parentheses needed?
```suggestion
        new_var = D_80C0B0F8[i] * -80.0f; // temp seems to actually be needed for matching
```

File: src/overlays/actors/ovl_Bg_Iknin_Susceil/z_bg_iknin_susceil.c [hunk 4]
```diff
@@ -21,46 +35,228 @@ const ActorInit Bg_Iknin_Susceil_InitVars = {
...
+    this->dyna.actor.velocity.y *= 0.93f;
+    this->dyna.actor.world.pos.y += this->dyna.actor.velocity.y;
+    if (this->dyna.actor.world.pos.y <= this->dyna.actor.home.pos.y) {
+        func_80C0A86C(this, globalCtx, 4, 14, 1);
+        Actor_UnsetSwitchFlag(globalCtx, GET_SUSCEIL_SWITCHFLAG(this));
+        Audio_PlayActorSound2(&this->dyna.actor, NA_SE_EV_BIGWALL_BOUND);
+        func_80C0AC74(this);
+        return;
+    }
+    func_800B9010(&this->dyna.actor, NA_SE_EV_ICE_PILLAR_FALL - SFX_FLAG);
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Bg_Iknin_Susceil/z_bg_iknin_susceil.c:161]:
```suggestion
        func_80C0AC74(this);
    } else {
        func_800B9010(&this->dyna.actor, NA_SE_EV_ICE_PILLAR_FALL - SFX_FLAG);
    }
```

File: src/overlays/actors/ovl_Bg_Iknin_Susceil/z_bg_iknin_susceil.c [hunk 5]
```diff
@@ -21,46 +35,228 @@ const ActorInit Bg_Iknin_Susceil_InitVars = {
...
+    this->dyna.actor.velocity.y *= 0.98f;
+    if (Math_SmoothStepToF(&this->dyna.actor.world.pos.y, this->dyna.actor.home.pos.y + 365.0f, 0.5f,
+                           this->dyna.actor.velocity.y, 1.0f) < 0.1f) {
+        func_80C0A86C(this, globalCtx, 1, 0xE, 3);
+        ActorCutscene_Stop(this->dyna.actor.cutscene);
+        func_80C0AB14(this);
+        return;
+    }
+    func_800B9010(&this->dyna.actor, NA_SE_EV_ICE_PILLAR_RISING - SFX_FLAG);
+}
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Bg_Iknin_Susceil/z_bg_iknin_susceil.c:204]:
```suggestion
        func_80C0AB14(this);
    } else {
        func_800B9010(&this->dyna.actor, NA_SE_EV_ICE_PILLAR_RISING - SFX_FLAG);
    }
}
```

**Review by AngheloAlf — COMMENTED** [review]
Good job!
Just some stylistic changes.

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**